### ISSUE TITLE
NavigationItem has different options in Figma between orientations but in code it does not

### ISSUE THREAD

**Issue by joshwooding** [jpmorganchase/salt-ds#3928]
### Package name(s)

Core (@salt-ds/core)

### Package version(s)

latest

### Description

NavigationItem in code allows all options on both orientations, Figma does not.

Figma currently doesn't have a horizontal item without expansion. However this is possible in code. This will also affect the App Header pattern in Figma.

Navigation Item in Figma makes hover and expanded a combined option rather than separate states.

### Operating system

- [ ] macOS
- [ ] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### Browser

- [ ] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

### Are you a JPMorgan Chase & Co. employee?

- [ ] I am an employee of JPMorgan Chase & Co.

**Comment by mark-tate** [comment]
@bhoppers2008 needs a designer to have a look at this ?

**Comment by bhoppers2008** [comment]
@yunjungyeh please review if this is still required for Salt DS Figma.

**Comment by yunjungyeh** [comment]
@bhoppers2008 Yes it's required, @joshwooding has updated more details in Description

**Comment by mark-tate** [comment]
NavigationItem being updated, so the spec will be replaced/updated.
